lib/parser/objects.ts

```diff
@@ -1068,6 +1068,11 @@ export function* scope(startObject: ObjectBase) {
     if (current instanceof OPackageBody && current.correspondingPackage) {
       yield current.correspondingPackage;
     }
+    if (implementsIHasUseClause(current)) {
+      for (const packages of current.packageDefinitions) {
+        yield packages;
+      }
+    }
     if (current.parent instanceof OFile) {
```

Comment:
I really like this change. But this is manually done at other points in the code, doing only this probably creates duplicated warnings/errors or?

Reply:
I haven't tested for duplicate messages, you are probably right.
We could change this PR to use the scope generator everywhere? Or should we create another PR which migrates everything to the scope generator and postpone the merge of this PR to after the scope migration PR?